contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -397,6 +397,55 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
+   void system_contract::denynames( std::vector<name> patterns ) {
+      require_auth( get_self() );
+
+      check(!patterns.empty(), "Empty list of blacklisted name patterns provided");
+      check(patterns.size() <= 512, "Cannot provide more than 512 patterns in one action call");
```

Comment:
Should we verify the patterns such that they only contain valid chars allowed by name?

Reply:
The pattern is passed as a name so the check was already made.